jax/_src/api.py

```diff
@@ -1575,58 +1575,62 @@ def _mapped_axis_size(tree, vals, dims, name, *, kws=False):
         f"containing an array, got empty *args={args} and **kwargs={kwargs}"
     )
 
-  def _get_axis_size(name: str, shape: Tuple[int, ...], axis: int):
+  def _get_axis_size(name: str,
+                     shape: Tuple[Union[int, core.BInt, core.Tracer], ...],
```

Comment:
The changes in this function are
1. change the type annotation on the `shape` parameter to include `BInt` and `Tracer`
2. change the `try`/`except` to an `if`/`else`
3. dedent / wrap some text

That is, the changes here are pretty trivial.